mwdb/resources/remotes.py

```diff
@@ -0,0 +1,457 @@
+from flask_restful import Resource
+from flask import g
+import requests
+from mwdb.core.plugins import hooks
+from werkzeug.exceptions import BadRequest, Conflict, NotFound, Forbidden
+from mwdb.core.capabilities import Capabilities
+from mwdb.core.config import app_config
+from mwdb.schema.remotes import APIRemotesListResponseSchema
+from mwdb.model import db, File, Config, TextBlob
+from mwdb.model.object import ObjectTypeConflictError
+from mwdb.schema.file import FileItemResponseSchema
+from mwdb.schema.config import ConfigItemResponseSchema
+from mwdb.schema.blob import BlobItemResponseSchema
+from tempfile import SpooledTemporaryFile
+from . import access_object, logger, requires_authorization
+
+
+class APIRemoteListResource(Resource):
+    @requires_authorization
+    def get(self):
+        """
+        ---
+        summary: Get list of configured remote names
+        description: |
+            Return a list of available remote names
+        security:
+            - bearerAuth: []
+        tags:
+            - remotes
+        responses:
+            200:
+                description: List of user configured remotes
+                content:
+                  application/json:
+                    schema: APIRemotesListResponseSchema
+        """
+        remotes = app_config.mwdb.remotes
+        schema = APIRemotesListResponseSchema()
+        return schema.dump({"remotes": remotes})
+
+
+def get_remote_api_data(remote_name):
+    if remote_name not in app_config.mwdb.remotes:
+        raise NotFound("Unknown remote instance name")
+
+    remote_url = app_config.get_key(f"remote:{remote_name}", "url")
+    api_key = app_config.get_key(f"remote:{remote_name}", "api_key")
+    return remote_url, api_key
+
+
+def map_remote_api_error(response):
+    if response.status_code == 200:
+        return None
+    elif response.status_code == 404:
+        raise NotFound("Remote object not found")
+    elif response.status_code == 403:
+        raise Forbidden("You are not permitted to perform this action on remote instance")
+    elif response.status_code == 409:
+        raise Conflict("Remote object already exists in remote instance and has different type")
+    else:
+        raise response.raise_for_status()
```

Comment:
It is enough to just call that function
```suggestion
        response.raise_for_status()
```